internal/sink/sink.go

```diff
@@ -39,15 +39,20 @@ type WebhookV1 struct {
 	// clientMiddleware func(http.Client) http.Client
 }
 
-func NewWebhookV1(s *sender) {
-	v1 := &WebhookV1{
-		id:               s.id,
-		deliveryInterval: s.deliveryInterval,
-		deliveryRetries:  s.deliveryRetries,
-		logger:           s.logger,
+func NewSink(c Config, logger *zap.Logger, id string, version int) Sink {
```

Comment:
based on my other comment https://github.com/xmidt-org/caduceus/pull/487/files#r1586542444
let's add the same `switch` here, otherwise version (and what it means) may not be obvious to anyone other than us
```
switch v := l.(type) {
	case *ListenerV1:
```

Reply:
if we pass the listener then we can use `l.GetId()` instead of passing in the `id`